### PR TITLE
Allow unlimited headers

### PR DESCRIPTION
**Pull request by rdubrock**
Allow unlimited headers in http requests

### CODE REVIEW

File: src/components/NameValueInput/NameValueInput.tsx [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
   disabled?: boolean;
   label: string;
   validateName?: (name: string) => string | undefined;
   validateValue?: (value: string) => string | undefined;
 }
 
+function getErrors(errors: FieldErrorsImpl<CheckFormValues>, name: NameValueName) {
+  switch (name) {
+    case 'settings.http.headers': {
+      const cast = errors as FieldErrorsImpl<CheckFormValuesHttp>;
```
Comment (rdubrock) [inline src/components/NameValueInput/NameValueInput.tsx:22]: I was fighting with this for a long time. I don't get why the new union types won't allow me to say `CheckFormValues` _might_ have HTTP on it. Is there a way to say "I have a form that might be of any check type, but if it's HTTP I would like to use the HTTP key"? Casting always feels icky.

@ckbedwell

File: src/components/http/HttpSettings.tsx [hunk 2]
```diff
@@ -211,7 +211,6 @@ export const HttpSettingsForm = ({ isEditor }: Props) => {
               name="settings.http.headers"
               disabled={!isEditor}
               label="header"
-              limit={10}
               validateName={validateHTTPHeaderName}
               validateValue={validateHTTPHeaderValue}
```
Comment (peterschretlen) [inline src/components/http/HttpSettings.tsx:215]: There is also a proxy header limit of 10 below - should we remove that one too?

Comment (rdubrock) [reply src/components/http/HttpSettings.tsx:215]: Yeah good call

File: src/components/NameValueInput/NameValueInput.tsx [hunk 3]
```diff
@@ -1,39 +1,54 @@
...
+  }
+}
+
 export const NameValueInput = ({ name, disabled, limit, label, validateName, validateValue }: Props) => {
   const {
     register,
     control,
     formState: { errors },
-  } = useFormContext(); // todo: type correctly
+  } = useFormContext<CheckFormValues>();
```
[the remark this answers is not in the document]
Comment (rdubrock) [reply src/components/NameValueInput/NameValueInput.tsx:39]: Any reason not to just union the two options? I feel like this component doesn't need to care, it just needs to know there's a `label` field

**Review by ckbedwell — APPROVED** [review]

**Pull request merged**